src/planner.ts

```diff
@@ -283,9 +280,15 @@ export class Planner {
                 }
                 ret = state.length;
 
+                const topNode = nextDeadSlot.peek();
+
                 // Is there a spare state slot?
-                if(nextDeadSlot.peek().dies <= i) {
-                    ret = nextDeadSlot.pop().slot;
+                if(typeof topNode !== "undefined" && topNode.dies <= i) {
+                    const extractedTopNode = nextDeadSlot.pop();
+
+                    if (extractedTopNode) {
+                        ret = extractedTopNode?.slot;
```

Comment:
Is it necessary to have both the conditional and the .? ? It seems like either one should do.

Reply:
Was causing a warning otherwise